src/constructs/aws/SinglePageApp.ts

```diff
@@ -0,0 +1,57 @@
+import * as cloudfront from "@aws-cdk/aws-cloudfront";
+import { FunctionEventType } from "@aws-cdk/aws-cloudfront";
+import type { Construct as CdkConstruct } from "@aws-cdk/core";
+import type { AwsProvider } from "@lift/providers";
+import type { FromSchema } from "json-schema-to-ts";
+import { getCfnFunctionAssociations } from "../../utils/getDefaultCfnFunctionAssociations";
+import { COMMON_STATIC_WEBSITE_DEFINITION, StaticWebsiteAbstract } from "./abstracts/StaticWebsiteAbstract";
+
+const SINGLE_PAGE_APP_DEFINITION = { ...COMMON_STATIC_WEBSITE_DEFINITION } as const;
+
+type Configuration = FromSchema<typeof SINGLE_PAGE_APP_DEFINITION>;
+
+export class SinglePageApp extends StaticWebsiteAbstract {
+    public static type = "single-page-app";
+    public static schema = SINGLE_PAGE_APP_DEFINITION;
+
+    constructor(
+        scope: CdkConstruct,
+        protected readonly id: string,
+        protected readonly configuration: Configuration,
+        protected readonly provider: AwsProvider
+    ) {
+        super(scope, id, configuration, provider);
+
+        const cfnDistribution = this.distribution.node.defaultChild as cloudfront.CfnDistribution;
+        const viewerRequestFunction = this.createViewerRequestCloudfrontFunction();
+
+        const defaultBehaviorFunctionAssociations = getCfnFunctionAssociations(cfnDistribution);
+
+        cfnDistribution.addOverride("Properties.DistributionConfig.DefaultCacheBehavior.FunctionAssociations", [
+            ...defaultBehaviorFunctionAssociations,
+            { EventType: FunctionEventType.VIEWER_REQUEST, FunctionARN: viewerRequestFunction.functionArn },
+        ]);
+    }
+
+    public createViewerRequestCloudfrontFunction(): cloudfront.Function {
+        /**
+         * CloudFront function that redirects nested paths to /index.html and
+         * let static files pass.
+         */
+        const code = `var REDIRECT_REGEX = /^[^.]+$|\\.(?!(css|gif|ico|jpg|js|png|txt|svg|woff|woff2|ttf|map|json)$)([^.]+$)/;
+
+function handler(event) {
+    var uri = event.request.uri;
+    var isUriToRedirect = REDIRECT_REGEX.test(uri);
+
+    if (isUriToRedirect) event.request.uri = "/index.html";
+
+    return event.request;
+}`;
+
+        return new cloudfront.Function(this, "ViewerRequestFunction", {
+            functionName: `${this.provider.stackName}-${this.provider.region}-${this.id}-viewerRequest`,
```

Comment:
Same here, if we can simplify and minimize diffs that's great:

```suggestion
        return new cloudfront.Function(this, "RequestFunction", {
            functionName: `${this.provider.stackName}-${this.provider.region}-${this.id}-request`,
```